TheBackBenchers/Helper.cs

```diff
@@ -0,0 +1,282 @@
+        }
+
+        /// <summary>
+        /// Function to list all blobs inside a directory also
+        /// stores the ouput in lists if required
+        /// </summary>
+        public async Task ListBlobs(DataLakeFileSystemClient fileSystemClient, string directoryName,
+            bool recursive, bool storeOutput)
+        {
+            IAsyncEnumerator<PathItem> enumerator =
+                fileSystemClient.GetPathsAsync(directoryName, recursive).GetAsyncEnumerator();
+
+            await enumerator.MoveNextAsync();
+
+            PathItem item = enumerator.Current;
+
+            while (item != null)
+            {
+                Console.WriteLine(item.Name);
+
+                // Store directories
+                if (storeOutput == true && item.IsDirectory == true)
+                {
+                    this.directoryList.Add(item.Name);
+                }
+
+                // Store files
+                if (storeOutput == true && item.IsDirectory == false)
+                {
+                    this.fileList.Add(item.Name);
+                }
+
+                if (!await enumerator.MoveNextAsync())
+                {
+                    break;
+                }
+
+                item = enumerator.Current;
+            }
+        }
+
+        /// <summary>
+        /// Get a service client for ADLS Gen2 and use AAD token for authorization
+        /// </summary>
+        public DataLakeServiceClient GetDataLakeServiceClient(string accountName, string clientID, 
+            string clientSecret, string tenantID, DataLakeClientOptions options = null)
+        {
+            TokenCredential credential = new ClientSecretCredential(
+                tenantID, clientID, clientSecret, new TokenCredentialOptions());
+
+            string dfsUri = "https://" + accountName + ".dfs.core.windows.net";
+
+            if (options == null)
+            {
+                return new DataLakeServiceClient(new Uri(dfsUri), credential);
+            }
+            else
+            {
+                return new DataLakeServiceClient(new Uri(dfsUri), credential, options);
+            }
+        }
+
+        /// <summary>
+        /// Get a service client for ADLS Gen2 [Blob endpoint] and use AAD token for authorization
+        /// </summary>
+        public BlobServiceClient GetBlobServiceClient(string accountName, string clientID,
+            string clientSecret, string tenantID, BlobClientOptions options = null)
+        {
+            TokenCredential credential = new ClientSecretCredential(
+                tenantID, clientID, clientSecret, new TokenCredentialOptions());
+
+            string dfsUri = "https://" + accountName + ".blob.core.windows.net";
+
+            if (options == null)
+            {
+                return new BlobServiceClient(new Uri(dfsUri), credential);
+            }
+            else
+            {
+                return new BlobServiceClient(new Uri(dfsUri), credential, options);
+            }

```

Comment:
probably just best leaving it as line 215, because passing null for options would be the equivalent as the other constructor anyways